### PR TITLE
Add simple authentication

### PR DESCRIPTION
**Pull request by corroded**
This has been just me trying to learn authentication with Phoenix haha. Will try to do some simple styles next so we can get rid of the Phoenix views boiler plates.

I currently have the keys for GITHUB_CLIENT_ID/GITHUB_CLIENT_SECRET but can pass it on if it works.

### CODE REVIEW

File: config/prod.exs [hunk 1]
```diff
@@ -13,7 +13,7 @@ use Mix.Config
 # which you typically run after static files are built.
 config :mantra, Mantra.Endpoint,
   http: [port: {:system, "PORT"}],
-  url: [scheme: "https", host: "mangtra.herokuapp.com", port: 443],
+  url: [scheme: "https", host: "pacific-plains-72253.herokuapp.com", port: 443],
```
Comment (SheepD) [inline config/prod.exs]: think we should move the host value to an env variable instead?

File: config/config.exs [hunk 2]
```diff
@@ -17,6 +17,24 @@ config :mantra, Mantra.Endpoint,
   pubsub: [name: Mantra.PubSub,
            adapter: Phoenix.PubSub.PG2]
 
+config :ueberauth, Ueberauth,
+  providers: [
+    github: {Ueberauth.Strategy.Github, []}
+  ]
+
+config :ueberauth, Ueberauth.Strategy.Github.OAuth,
+  client_id: System.get_env("GITHUB_CLIENT_ID"),
```
Comment (SheepD) [inline config/config.exs:26]: how do you manage env variables locally? i got spoiled by figaro on ruby hahaha.

File: config/prod.exs (hunk 1 above)
Comment (corroded) [reply config/prod.exs]: good idea! (and nice catch. thought I fixed them all)

will do!

File: config/config.exs (hunk 2 above)
Comment (corroded) [reply config/config.exs:26]: I used `dotenv` in ruby but now that I'm trying diff languages I use direnv (`brew install direnv`). will check your directory's .envrc for env settings. I can add a sample .envrc to this PR

**Comment by corroded** [comment]
Sorry I thought force pushing would purge those changes haha. Anyway if you're okay with this then this is good to merge 👍  I'll setup Heroku to have those env variables

**Pull request merged**